src/gluonts/trainer/_base.py

```diff
@@ -260,6 +254,14 @@ def base_path() -> str:
                                 else:
                                     loss = output
 
+                            # only invoke the following during the first forward pass
+                            if it == 1:
```

Comment:
I think this would fail as 'it' is not comparable and that you want to make this test on 'batch_no'. Otherwise the change looks good to me.